go/vt/discovery/healthcheck.go

```diff
@@ -467,10 +468,16 @@ func (hc *HealthCheckImpl) deleteTablet(tablet *topodata.Tablet) {
 				continue
 			}
 			delete(ths, tabletAlias)
-			// delete from healthy list
+
 			healthy, ok := hc.healthy[key]
 			if ok && len(healthy) > 0 {
-				hc.recomputeHealthy(key)
+				if tabletType == topodata.TabletType_PRIMARY {
```

Comment:
Thinking about this some more..
- we are deleting a tablet. if that tablet's type is not PRIMARY, we should not even "recompute the healthy primary"
- even if that tablet's type is PRIMARY, why should we iterate over the healthData for PRIMARY? we know that at any point there should only be one "healthy" primary. i.e `healthy[PRIMARY]` should have a size of either 0 or 1. If the tablet that we are deleting is THE current healthy primary, we should just set `healthy[PRIMARY]` to an empty slice. If it is not, we should do nothing.

It seems to me that the problem with the original implementation of this defer func was that it was calling `recomputeHealthy` for PRIMARY type (which was never done before this defer func was added) instead of handling PRIMARY separately. This caused it to violate the assumption that there is at most one `healthy` PRIMARY at any point in time.

Reply:
I think you're absolutely right. I simplified this changeset by simply performing the same logic as in `updateHealth` for `PRIMARY` tablet types, and calling `recomputeHealth` for the other types.